File: frontend/src/utils/agentConstants.js
/**
 * Agent constants for consistent ID mapping across the frontend
 */

// Agent IDs
export const EMAIL_AGENT_ID = 1;
export const IDEAS_AGENT_ID = 8;
export const QUOTE_AGENT_ID = 5;

// Agent Names
export const EMAIL_AGENT_NAME = "Email Agent";
export const IDEAS_AGENT_NAME = "Ideas Agent";
export const QUOTE_AGENT_NAME = "Quote Agent";

// Agent ID to Name mapping
export const AGENT_ID_TO_NAME = {
  [EMAIL_AGENT_ID]: EMAIL_AGENT_NAME,
  [IDEAS_AGENT_ID]: IDEAS_AGENT_NAME,
  [QUOTE_AGENT_ID]: QUOTE_AGENT_NAME,
};

// Agent Name to ID mapping
export const AGENT_NAME_TO_ID = {
  [EMAIL_AGENT_NAME]: EMAIL_AGENT_ID,
  [IDEAS_AGENT_NAME]: IDEAS_AGENT_ID,
  [QUOTE_AGENT_NAME]: QUOTE_AGENT_ID,
};

/**
 * Get agent name by ID
 * @param {number} agentId - The agent ID
 * @returns {string} The agent name
 */
export const getAgentName = (agentId) => {
  return AGENT_ID_TO_NAME[agentId] || `Agent ${agentId}`;
};

/**
 * Get agent ID by name
 * @param {string} agentName - The agent name
 * @returns {number} The agent ID
 */
export const getAgentId = (agentName) => {
  return AGENT_NAME_TO_ID[agentName] || 0;
};

/**
 * Check if agent is Email Agent
 * @param {number} agentId - The agent ID
 * @returns {boolean} True if Email Agent
 */
export const isEmailAgent = (agentId) => {
  return agentId === EMAIL_AGENT_ID;
};

/**
 * Check if agent is Ideas Agent
 * @param {number} agentId - The agent ID
 * @returns {boolean} True if Ideas Agent
 */
export const isIdeasAgent = (agentId) => {
  return agentId === IDEAS_AGENT_ID;
};

/**
 * Check if agent is Quote Agent
 * @param {number} agentId - The agent ID
 * @returns {boolean} True if Quote Agent
 */
export const isQuoteAgent = (agentId) => {
  return agentId === QUOTE_AGENT_ID;
}; 

File: backend/app/utils/agent_constants.py
"""
Agent constants for consistent ID mapping across the application
"""

# Agent IDs
EMAIL_AGENT_ID = 1
QUEST_AGENT_ID = 2
QUERY_AGENT_ID = 3
IDEAS_AGENT_ID = 8
QUOTE_AGENT_ID = 5
AGENT_X_ID = 'X'
AGENT_Y_ID = 'Y'
AGENT_Z_ID = 'Z'

# Agent Names
EMAIL_AGENT_NAME = "Email Agent"
QUEST_AGENT_NAME = "Quest Agent"
QUERY_AGENT_NAME = "Query Agent"
IDEAS_AGENT_NAME = "Ideas Agent"
QUOTE_AGENT_NAME = "Quote Agent"
AGENT_X_NAME = "Agent X"
AGENT_Y_NAME = "Agent Y"
AGENT_Z_NAME = "Agent Z"

# Agent ID to Name mapping
AGENT_ID_TO_NAME = {
    EMAIL_AGENT_ID: EMAIL_AGENT_NAME,
    QUEST_AGENT_ID: QUEST_AGENT_NAME,
    QUERY_AGENT_ID: QUERY_AGENT_NAME,
    IDEAS_AGENT_ID: IDEAS_AGENT_NAME,
    QUOTE_AGENT_ID: QUOTE_AGENT_NAME,
    AGENT_X_ID: AGENT_X_NAME,
    AGENT_Y_ID: AGENT_Y_NAME,
    AGENT_Z_ID: AGENT_Z_NAME,
}

# Agent Name to ID mapping
AGENT_NAME_TO_ID = {
    EMAIL_AGENT_NAME: EMAIL_AGENT_ID,
    QUEST_AGENT_NAME: QUEST_AGENT_ID,
    QUERY_AGENT_NAME: QUERY_AGENT_ID,
    IDEAS_AGENT_NAME: IDEAS_AGENT_ID,
    QUOTE_AGENT_NAME: QUOTE_AGENT_ID,
    AGENT_X_NAME: AGENT_X_ID,
    AGENT_Y_NAME: AGENT_Y_ID,
    AGENT_Z_NAME: AGENT_Z_ID,
}

def get_agent_name(agent_id) -> str:
    """Get agent name by ID"""
    return AGENT_ID_TO_NAME.get(agent_id, f"Agent {agent_id}")

def get_agent_id(agent_name: str):
    """Get agent ID by name"""
    return AGENT_NAME_TO_ID.get(agent_name, 0)

def is_email_agent(agent_id) -> bool:
    """Check if agent is Email Agent"""
    return agent_id == EMAIL_AGENT_ID

def is_quest_agent(agent_id) -> bool:
    """Check if agent is Quest Agent"""
    return agent_id == QUEST_AGENT_ID

def is_query_agent(agent_id) -> bool:
    """Check if agent is Query Agent"""
    return agent_id == QUERY_AGENT_ID

def is_ideas_agent(agent_id) -> bool:
    """Check if agent is Ideas Agent"""
    return agent_id == IDEAS_AGENT_ID

def is_quote_agent(agent_id) -> bool:
    """Check if agent is Quote Agent"""
    return agent_id == QUOTE_AGENT_ID

def is_agent_x(agent_id) -> bool:
    """Check if agent is Agent X"""
    return agent_id == AGENT_X_ID

def is_agent_y(agent_id) -> bool:
    """Check if agent is Agent Y"""
    return agent_id == AGENT_Y_ID

def is_agent_z(agent_id) -> bool:
    """Check if agent is Agent Z"""
    return agent_id == AGENT_Z_ID 